lib/utils/api_utils.dart

```diff
@@ -0,0 +1,148 @@
+      {Map<String, String> headers, dynamic body}) async {
+    try {
+      final response = await client.put(
+        uri,
+        headers: headers,
+        body: jsonEncode(body),
+      );
+      final jsonResponse = ApiUtils.jsonResponse(response);
+      return jsonResponse;
+    } on SocketException catch (e) {
+      print(e.message);
+      throw Failure(Constants.NO_INTERNET_CONNECTION);
+    } on HttpException catch (e) {
+      print(e.message);
+      throw Failure(Constants.HTTP_EXCEPTION);
+    }
+  }
+
+  /// Returns JSON PATCH response
+  static Future<dynamic> patch(String uri,
+      {Map<String, String> headers, dynamic body}) async {
+    try {
+      final response =
+          await client.patch(uri, headers: headers, body: jsonEncode(body));
+      final jsonResponse = ApiUtils.jsonResponse(response);
+      return jsonResponse;
+    } on SocketException catch (e) {
+      print(e.message);
+      throw Failure(Constants.NO_INTERNET_CONNECTION);
+    } on HttpException catch (e) {
+      print(e.message);
+      throw Failure(Constants.HTTP_EXCEPTION);
+    }
+  }
+
+  /// Returns JSON DELETE response
+  static Future<dynamic> delete(String uri,
+      {Map<String, String> headers}) async {
+    try {
+      final response = await http.delete(uri, headers: headers);
+      final jsonResponse = ApiUtils.jsonResponse(response);
+      return jsonResponse;
+    } on SocketException catch (e) {
+      print(e.message);
+      throw Failure(Constants.NO_INTERNET_CONNECTION);
+    } on HttpException catch (e) {
+      print(e.message);
+      throw Failure(Constants.HTTP_EXCEPTION);
+    }
+  }
+
+  static dynamic jsonResponse(http.Response response) {
+    switch (response.statusCode) {
+      case 200:
+      case 201:
+      case 202:
+      case 204:
+        var responseJson =
+            response.body == '' ? {} : json.decode(response.body);
+        print(responseJson);
+        return responseJson;
+      case 400:
+        print(response.body);
+        throw BadRequestException(response.body);
+      case 401:
+        print(response.body);
+        throw UnauthorizedException(response.body);
+      case 403:
+        print(response.body);
+        throw ForbiddenException(response.body);
+      case 404:
+        print(response.body);
+        throw NotFoundException(response.body);
+      case 409:
+        print(response.body);
+        throw ConflictException(response.body);
+      case 422:
+        print(response.body);
+        throw UnprocessableIdentityException(response.body);
+      case 500:
+        print(response.body);
+        throw InternalServerErrorException(response.body);
+      case 503:
+        print(response.body);
+        throw ServiceUnavailableException(response.body);
+      default:
+        print(response.body);
+        throw FetchDataException(
+          'Error occured while Communication with Server with StatusCode : ${response.statusCode}',
```

Comment:
@Nitish145 typo

Reply:
Oops! Will update..

Reply:
`Error occured while communicating with server with status code`.. is this ok ?